outputs/policyreport.go

```diff
@@ -0,0 +1,316 @@
+package outputs
+
+import (
+	"context"
+	"fmt"
+	"log"
+
+	"github.com/DataDog/datadog-go/statsd"
+	"github.com/falcosecurity/falcosidekick/types"
+	"github.com/google/uuid"
+	"github.com/kubernetes-sigs/wg-policy-prototypes/policy-report/kube-bench-adapter/pkg/apis/wgpolicyk8s.io/v1alpha2"
+	wgpolicy "github.com/kubernetes-sigs/wg-policy-prototypes/policy-report/kube-bench-adapter/pkg/apis/wgpolicyk8s.io/v1alpha2"
+	crdClient "github.com/kubernetes-sigs/wg-policy-prototypes/policy-report/kube-bench-adapter/pkg/generated/v1alpha2/clientset/versioned"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"k8s.io/client-go/util/retry"
+)
+
+//for n namespace specific reports with their count
+type pr struct {
+	report *wgpolicy.PolicyReport
+	count  int
+}
+
+const (
+	clusterPolicyReportBaseName = "falco-cluster-policy-report-"
+	policyReportSource          = "Falco"
+	highpri                     = "high"
+)
+
+var (
+	failbound int
+	//count for cluster policy report
+	repcount int
+	//slice of policyreports and their counts(type pr)
+	polreports = make(map[string]*pr)
+	//for cluster policyreport
+	report *wgpolicy.ClusterPolicyReport = &wgpolicy.ClusterPolicyReport{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: clusterPolicyReportBaseName,
+		},
+		Summary: v1alpha2.PolicyReportSummary{
+			Fail: 0,
+			Warn: 0, //to-do
+		},
+	}
+)
+
+func NewPolicyReportClient(config *types.Configuration, stats *types.Statistics, promStats *types.PromStatistics, statsdClient, dogstatsdClient *statsd.Client) (*Client, error) {
+	clientConfig, err := rest.InClusterConfig()
+	if err != nil {
+		clientConfig, err = clientcmd.BuildConfigFromFlags("", config.PolicyReport.Kubeconfig)
+		if err != nil {
+			fmt.Printf("[ERROR] :unable to load kube config file: %v", err)
+		}
+	}
+	crdclient, err := crdClient.NewForConfig(clientConfig)
+	if err != nil {
+		return nil, err
+	}
+	report.ObjectMeta.Name += uuid.NewString()[:8]
+	return &Client{
+		OutputType:      "PolicyReport",
+		Config:          config,
+		Stats:           stats,
+		PromStats:       promStats,
+		StatsdClient:    statsdClient,
+		DogstatsdClient: dogstatsdClient,
+		Crdclient:       crdclient,
+	}, nil
+}
+
+// CreateReport creates PolicyReport/ClusterPolicyReport Resource in Kubernetes
+func (c *Client) CreateReport(falcopayload types.FalcoPayload) {
+	failbound = c.Config.PolicyReport.FailThreshold
+	r, namespaceScoped := newResult(falcopayload)
+	if namespaceScoped != "" {
+		forPolicyReports(c, namespaceScoped, r)
+	} else {
+		forClusterPolicyReport(c, r)
+	}
+}
+
+//newResult creates a new entry for Reports
+func newResult(FalcoPayload types.FalcoPayload) (c *wgpolicy.PolicyReportResult, namespaceScoped string) {
+	namespaceScoped = "" // decision variable to increment for policyreport and clusterpolicyreport //to do //false for clusterpolicyreport
```

Comment:
no need to init, this is implicit in Go as you declare a string in output variables